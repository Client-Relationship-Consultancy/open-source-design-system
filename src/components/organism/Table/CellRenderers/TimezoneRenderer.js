import React from "react"
import PropTypes from "prop-types"
import TimezoneSelect from "../../../molecules/TimezoneSelect"

export default class TimezoneRenderer extends React.Component {
  state = {
    value: "",
  }

  handleChange = ({ value }) => this.setState({ value })

  getValue = () => this.state.value

  render() {
    return (
      <TimezoneSelect
        id="timezone"
        name="timezone"
        onChange={this.handleChange}
        isDisabled={this.props.isDisabled}
        useFormik={false}
      />
    )
  }
}

TimezoneRenderer.propTypes = {
  isDisabled: PropTypes.bool,
}

TimezoneRenderer.displayName = "TimezoneRenderer"
