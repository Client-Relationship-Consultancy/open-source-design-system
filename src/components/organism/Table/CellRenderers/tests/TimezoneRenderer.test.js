import React from "react"
import { shallow } from "enzyme"
import "jest-styled-components"
import TimezoneRenderer from "../TimezoneRenderer"

describe("Test the timezone Renderer", () => {
  let component
  beforeEach(() => {
    component = shallow(<TimezoneRenderer />)
  })

  it("matches last snapshot", () => {
    expect(component).toMatchSnapshot()
  })

  it("Sets the selected value on handleChange", () => {
    const newOption = "option 1"
    component.instance().handleChange({ value: newOption })
    expect(component.instance().getValue()).toEqual(newOption)
  })
})
