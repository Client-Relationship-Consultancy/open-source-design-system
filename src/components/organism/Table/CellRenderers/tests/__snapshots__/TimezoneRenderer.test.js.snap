// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the timezone Renderer matches last snapshot 1`] = `
<TimezoneSelect
  id="timezone"
  isClearable={true}
  name="timezone"
  onChange={[Function]}
  placeholder="Type to select a timezone..."
  useFormik={false}
/>
`;
